hw-model/c-binding/src/caliptra_model.rs

```diff
@@ -0,0 +1,149 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_emu_bus::Bus;
+use caliptra_hw_model::{DefaultHwModel, HwModel, InitParams};
+use std::ffi::*;
+use std::slice;
+
+use caliptra_emu_types::RvSize;
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct caliptra_model {
+    _unused: [u8; 0],
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct caliptra_buffer {
+    pub data: *const u8,
+    pub len: usize,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct caliptra_model_init_params {
+    pub rom: caliptra_buffer,
+    pub dccm: caliptra_buffer,
+    pub iccm: caliptra_buffer,
+}
+
+pub const CALIPTRA_MODEL_STATUS_OK: c_int = 0;
+
+/// # Safety
+#[no_mangle]
+pub unsafe extern "C" fn caliptra_model_init_default(
+    params: caliptra_model_init_params,
+    model: *mut *mut caliptra_model,
+) -> c_int {
+    // Parameter check
+    assert!(!model.is_null());
+    // Generate Model and cast to caliptra_model
+    *model = Box::into_raw(Box::new(
+        caliptra_hw_model::new_unbooted(InitParams {
+            rom: slice::from_raw_parts(params.rom.data, params.rom.len),
+            dccm: slice::from_raw_parts(params.dccm.data, params.dccm.len),
+            iccm: slice::from_raw_parts(params.iccm.data, params.iccm.len),
+            ..Default::default()
+        })
+        .unwrap(),
+    )) as *mut caliptra_model;
+
+    CALIPTRA_MODEL_STATUS_OK
+}
+
+/// # Safety
+#[no_mangle]
+pub unsafe extern "C" fn caliptra_model_destroy(model: *mut caliptra_model) {
+    // Parameter check
+    assert!(!model.is_null());
+
+    // This will force model to be freed
+    let _ = Box::from_raw(model);
```

Comment:
This might be a bit clearer as `drop(Box::from_raw(model))`.

Reply:
done